heron/common/src/cpp/metrics/metricsmgr-client.cpp

```diff
@@ -63,6 +63,12 @@ void MetricsMgrClient::HandleClose(NetworkErrorCode) {
 
 void MetricsMgrClient::ReConnect() { Start(); }
 
+void MetricsMgrClient::SetPublisherPort(const sp_int32 _port) {
+  if (0 == port_ && 0 != _port) {
```

Comment:
If the input _port is zero, which is not a actual port in use, it is not a valid input.
If the existing port_ in MetricsMgrClient is not zero, it should not be set to another value during MetricsMgrClient lifetime.